app/controllers/concerns/helpers.rb

```diff
@@ -52,16 +48,17 @@ def attr_display_str(obj, attrs_to_display)
       end
     end
 
-    def viewing_own_profile_while_logged_in?(user, current_user)
+    def viewing_own_profile_while_logged_in?(user)
       if user && current_user
-        user.id == current_user.id
+        user.id == @current_user.id
+      else
+        false
       end
     end
 
     def viewing_own_activity?(activity)

```

Comment:
I think we can just change this whole method to be this 

```activity.user == current_user``` 

Might not even need to be a helper method, as it is very concise and could live in a controller